packages/neutron-sdk/src/interchain_queries/v045/testing.rs

```diff
@@ -1144,9 +1145,70 @@ fn test_unbonding_delegations_reconstruct_from_hex() {
                         completion_time: Some(Timestamp::from_nanos(1704872903002248037)),
                         creation_height: 377,
                         initial_balance: Uint128::new(20_000),
-                    }
+                    },
                 ],
             }]
         }
     );
 }
+
+#[test]
+fn test_deconstruct_account_denom_balance_key() {
+    struct TestCase {
+        key: Vec<u8>,
+        expected_result: NeutronResult<(AddressBytes, String)>,
+    }
+
+    let testcases = vec![
+        TestCase {
+            key: create_account_denom_balance_key("addr1", "uatom").unwrap(),
+            expected_result: Ok((
+                "addr1".bytes().collect::<AddressBytes>(),
+                "uatom".to_string(),
+            )),
+        },
+        TestCase {
+            key: vec![],
+            expected_result: Err(NeutronError::AccountDenomBalanceKeyDeconstructionError(
+                "invalid key length".to_string(),
+            )),
+        },
+        TestCase {
+            key: vec![STAKING_PARAMS_KEY],
+            expected_result: Err(NeutronError::AccountDenomBalanceKeyDeconstructionError(
+                "first element is not 2".to_string(),
+            )),
+        },
+        TestCase {
+            key: vec![BALANCES_PREFIX],
+            expected_result: Err(NeutronError::AccountDenomBalanceKeyDeconstructionError(
+                "invalid key length".to_string(),
+            )),
+        },
+        TestCase {
+            key: vec![BALANCES_PREFIX, 10, 1],
+            expected_result: Err(NeutronError::AccountDenomBalanceKeyDeconstructionError(
+                "address length in key in invalid".to_string(),
+            )),
+        },
+        TestCase {
+            key: vec![BALANCES_PREFIX, 2, 1, 2],
+            expected_result: Err(NeutronError::AccountDenomBalanceKeyDeconstructionError(
+                "denom in key can't be empty".to_string(),
+            )),
+        },
+        TestCase {
+            key: vec![BALANCES_PREFIX, 2, 1, 2, 0, 159, 146, 150],
```

Comment:
could you please add some description to such magic values used in test cases?

Reply:
Fixed